ddl/backfilling_dispatcher.go

```diff
@@ -55,54 +64,54 @@ func (*backfillingDispatcherExt) OnTick(_ context.Context, _ *proto.Task) {
 }
 
 // OnNextSubtasksBatch generate batch of next stage's plan.
-func (h *backfillingDispatcherExt) OnNextSubtasksBatch(ctx context.Context, _ dispatcher.TaskHandle, gTask *proto.Task) ([][]byte, error) {
-	var globalTaskMeta BackfillGlobalMeta
-	if err := json.Unmarshal(gTask.Meta, &globalTaskMeta); err != nil {
+func (h *backfillingDispatcherExt) OnNextSubtasksBatch(ctx context.Context,
+	taskHandle dispatcher.TaskHandle,
+	gTask *proto.Task,
+) (taskMeta [][]byte, err error) {
+	var gTaskMeta BackfillGlobalMeta
+	if err := json.Unmarshal(gTask.Meta, &gTaskMeta); err != nil {
 		return nil, err
 	}
-	job := &globalTaskMeta.Job
-	var tblInfo *model.TableInfo
+
+	useExtStore := len(gTaskMeta.CloudStorageURI) > 0
+	defer func() {
+		// Only redact when the task is complete.
+		if len(taskMeta) == 0 && useExtStore {
+			redactCloudStorageURI(ctx, gTask, &gTaskMeta)
+			if err := taskHandle.UpdateTask(gTask.State, nil, dispatcher.RetrySQLTimes); err != nil {
+				logutil.Logger(ctx).Error("failed to UpdateTask", zap.Error(err))
```

Comment:
seems this changes nothing? @ywqzzy

Reply:
I will move this to framework in the next pr

Reply:
as discussed, we will move all task status change to framework later. cc @ywqzzy